Q2.cpp

```diff
@@ -0,0 +1,83 @@
+//CodeU: Q2
+//
+//  Created by Julia Semavina on 18.05.2017
+//
+//  problem specification:
+//      Implement an algorithm to find the kth to last element of a singly linked list
+//
+//  comments:
+//      list length is unknown
+//      if K is bigger than the size of the list the answer will be IMPOSSIBLE INPUT
+//      KthToLast function contains a solution
+//
+
+#include <iostream>
+#include <vector>
+
+
+struct Node {
+    char value;
+    Node* next;
+};
+
+Node* CreateSinglyLinkedList() {
+    Node *Head = new Node;
+    std::cin >> Head->value;
```

Comment:
Also, instead (or in addition to) the user entering the data, it may be good to have some code that creates test cases and compares the results against expected values, that way you don't have to manually enter them each time you want to check your code. A common pattern is:
void doAllTests() {
    doOneTest(input, expected output);
    ....
}